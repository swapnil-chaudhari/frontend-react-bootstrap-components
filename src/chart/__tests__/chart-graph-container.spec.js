import React from 'react';
import renderShallow from 'render-shallow';
import MockDate from 'mockdate';
import ChartGraphContainer from '../chart-graph-container';
import mockData from './chart-graph-mock-data.json';

describe('<ChartGraphContainer>', () => {
    describe('when it renders with data', () => {
        let component;
        const props = {
            ...mockData,
            hoverOnDate: () => {}
        };

        beforeAll(() => {
            MockDate.set('10/13/2017');
            component = renderShallow(
                <ChartGraphContainer { ...props } />
            ).output;
        });

        afterAll(() => {
            MockDate.reset();
        });

        test('has proper markup', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with loader', () => {
        let component;
        const props = {
            ...mockData,
            hoverOnDate: () => {},
            isLoading: true
        };

        beforeAll(() => {
            MockDate.set('10/13/2017');
            component = renderShallow(
                <ChartGraphContainer { ...props } />
            ).output;
        });

        afterAll(() => {
            MockDate.reset();
        });

        test('has proper markup with loader', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
