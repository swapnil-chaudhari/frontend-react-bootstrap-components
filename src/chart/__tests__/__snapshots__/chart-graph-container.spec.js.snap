// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartGraphContainer> when it renders with data has proper markup 1`] = `
<div
  className="ChartInteractive"
>
  <ChartContainer
    height={400}
  >
    <Graph
      circleRadius={6}
      circleYOffset={2}
      colors={
        Array [
          "#4B3DA3",
          "#419B86",
          "#026EAD",
        ]
      }
      data={
        Array [
          Object {
            "activity_datetime": "2017-10-06",
            "facebook": Object {
              "metric_value": 200,
            },
            "instagram": Object {
              "metric_value": 40,
            },
            "spotify": Object {
              "metric_value": 50,
            },
          },
          Object {
            "activity_datetime": "2017-10-08",
            "facebook": Object {
              "metric_value": 150,
            },
            "instagram": Object {
              "metric_value": 40,
            },
            "spotify": Object {
              "metric_value": 100,
            },
          },
          Object {
            "activity_datetime": "2017-10-13",
            "facebook": Object {
              "metric_value": 350,
            },
            "instagram": Object {
              "metric_value": 30,
            },
            "spotify": Object {
              "metric_value": 50,
            },
          },
        ]
      }
      days={7}
      height={400}
      hoverOnDate={[Function]}
      hoverSpotOffset={8}
      hoverType="accumulation"
      isLoading={false}
      leftTooltipOffset={3}
      noDataText="-"
      sourceTypes={
        Array [
          "facebook",
          "spotify",
          "instagram",
        ]
      }
      tooltipHeight={25}
      topTooltipOffset={15}
      xAxisDateFormat="MMM D"
      xAxisDateFormatHover="dddd, MMMM D"
      xAxisKey="activity_datetime"
      xAxisOffset={50}
      yAxisKey="metric_value"
      yAxisOffset={-25}
    />
  </ChartContainer>
</div>
`;

exports[`<ChartGraphContainer> when it renders with loader has proper markup with loader 1`] = `
<div
  className="ChartInteractive"
>
  <ChartContainer
    height={400}
  >
    <Graph
      circleRadius={6}
      circleYOffset={2}
      colors={
        Array [
          "#4B3DA3",
          "#419B86",
          "#026EAD",
        ]
      }
      data={
        Array [
          Object {
            "activity_datetime": "2017-10-06",
            "facebook": Object {
              "metric_value": 200,
            },
            "instagram": Object {
              "metric_value": 40,
            },
            "spotify": Object {
              "metric_value": 50,
            },
          },
          Object {
            "activity_datetime": "2017-10-08",
            "facebook": Object {
              "metric_value": 150,
            },
            "instagram": Object {
              "metric_value": 40,
            },
            "spotify": Object {
              "metric_value": 100,
            },
          },
          Object {
            "activity_datetime": "2017-10-13",
            "facebook": Object {
              "metric_value": 350,
            },
            "instagram": Object {
              "metric_value": 30,
            },
            "spotify": Object {
              "metric_value": 50,
            },
          },
        ]
      }
      days={7}
      height={400}
      hoverOnDate={[Function]}
      hoverSpotOffset={8}
      hoverType="accumulation"
      isLoading={true}
      leftTooltipOffset={3}
      noDataText="-"
      sourceTypes={
        Array [
          "facebook",
          "spotify",
          "instagram",
        ]
      }
      tooltipHeight={25}
      topTooltipOffset={15}
      xAxisDateFormat="MMM D"
      xAxisDateFormatHover="dddd, MMMM D"
      xAxisKey="activity_datetime"
      xAxisOffset={50}
      yAxisKey="metric_value"
      yAxisOffset={-25}
    />
  </ChartContainer>
  <div
    className="Graph-indicator-container"
  >
    <LoadingIndicator />
  </div>
</div>
`;
